guiEditor/src/diagram/workbench.tsx

```diff
@@ -80,19 +83,50 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
             }
         });
 
-        props.globalState.onPanObservable.add( () => {
-            this._forcePanning = true;
+        props.globalState.onPanObservable.add(() => {
+            this._forcePanning = ! this._forcePanning;
+            this._forceSelecting = false;
+            this._forceZooming = false;
+            if(!this._forcePanning) {
+                this.globalState.onSelectionObservable.notifyObservers();
+            }
+            else {
+                const canvas = document.getElementById("workbench-canvas") as HTMLCanvasElement;
+                canvas.style.cursor = "move";
+            }
         });
-        
-        props.globalState.onSelectionObservable.add( () => {
+
+        props.globalState.onSelectionObservable.add(() => {
+            this._forceSelecting = true;
             this._forcePanning = false;
+            this._forceZooming = false;
+            const canvas = document.getElementById("workbench-canvas") as HTMLCanvasElement;
+            canvas.style.cursor = "default"
         });
 
-        props.globalState.onZoomObservable.add( () => {
+        props.globalState.onZoomObservable.add(() => {
+
             this._forceZooming = !this._forceZooming;
+            this._forcePanning = false;
+            this._forceSelecting = false;
+            if(!this._forceZooming) {
+                this.globalState.onSelectionObservable.notifyObservers();
+            }
+            else {
+                const canvas = document.getElementById("workbench-canvas") as HTMLCanvasElement;

```

Comment:
if this never changes (and it doesn't, since it is an element with id), why not have a reference of the element during construction? This way you will save a few `getElementById` calls.